src/hammr/commands/image/image.py

```diff
@@ -491,14 +472,66 @@ def is_image_ready_to_publish(self, image, builder):
 
         return True
 
-
-
     def find_builder(self, image, template):
         for builder in template["builders"]:
-            format= self.get_image_format(builder)
-            if format!=2:
-                if image.format.name == format:
-                    return builder
+            if image.targetFormat.name == builder["type"]:
+                return builder
+
         return None
 
-        
\ No newline at end of file
+    def get_image_to_publish(self, builder, template, appliance, i):
+        printer.out(
+            "Publishing '" + builder["type"] + "' image (" + str(i) + "/" + str(len(template["builders"])) + ")")
+        images = self.api.Users(self.login).Appliances(appliance.dbId).Images.Getall(
+            Query="targetFormat.name=='" + builder["type"] + "'")
+        images = images.images.image
+        if images is None or len(images) == 0:
+            printer.out(
+                "No images found for the template '" + template["stack"]["name"] + "' with type: " + builder[
+                    "type"], printer.ERROR)
+            return 2
+
+        compliantImages = []
+        for image in images:
+            isOk = self.is_image_ready_to_publish(image, builder)
+            if isOk:
+                compliantImages.append(image)
+
+        if (len(compliantImages) == 0):
+            printer.out(
+                "No images found for the template '" + template["stack"]["name"] + "' with type: " + builder[
+                    "type"], printer.ERROR)
+            return 2
+        elif (len(compliantImages) == 1):
+            comliantImage = compliantImages[0]
+        else:
+            # TODO get the last created image
+            comliantImage = compliantImages[0]
+
+        return comliantImage
+
+    def get_account_to_publish(self, builder):
+        if not "account" in builder:
+            printer.out("Missing account section on builder: [" + builder["type"] + "]", printer.ERROR)
+            return 2
+        # Get all cloud account on the plateform (for the user)
+        accounts = self.api.Users(self.login).Accounts.Getall()
+        accounts = accounts.credAccounts.credAccount
+        if accounts is None or len(accounts) == 0:
+            printer.out("No accounts available on the plateform", printer.ERROR)
+            return 2
+        else:
+            for account in accounts:
+                if account.name == builder["account"]["name"]:
+                    # A hack to avoid a toDOM, toXML bug
```

Comment:
Could you please log an issue with the sample to reproduce the bug on Hammr bug tracker ?